src/librustc/middle/resolve_lifetime.rs

```diff
@@ -2066,27 +2070,78 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         }
     }
 
+    fn lint_implicit_lifetimes_in_segment(&mut self,
+                                          segment_ident: ast::Ident,
+                                          generic_args: &'tcx hir::GenericArgs,
+                                          lifetime_refs: &[&'tcx hir::Lifetime]) {
+        let num_implicit_lifetimes = lifetime_refs.iter()
+            .filter(|lt| lt.name.is_implicit()).count();
+        if num_implicit_lifetimes == 0 {
+            return;
+        }
+
+        let mut err = self.tcx.struct_span_lint_node(
+            lint::builtin::ELIDED_LIFETIMES_IN_PATHS,
+            lifetime_refs[0].id, // FIXME: HirIdify #50928
+            segment_ident.span,
+            &format!("implicit lifetime parameters in types are deprecated"),
+        );
+
+        if num_implicit_lifetimes == 1 {
```

Comment:
Why only the case of 1 implicit lifetime?

Reply:
It seems like the logic below would apply to any number of lifetimes, if we just generalized the string.